### ISSUE TITLE
"spacy train" command - questions about some arguments

### ISSUE THREAD

**Issue by CatarinaPC** [explosion/spaCy#4923]
I have a few questions regarding the training of a model using the [CLI](https://spacy.io/api/cli#train).
<br>
There is an argument on the `spacy train` command named `--n-iter` which is described as the number of iterations. 

1. Is that the same as number of epochs? 
2. Is that the maximum number of iterations/epochs or there is only a maximum when the argument `--n-early-stopping` is set?

<br>
There is also the argument `--use-gpu` described as:

> Whether to use GPU. Can be either 0, 1 or -1.

What does each value (0, 1 and -1) mean?
<br>

What do the arguments `--n_examples` and `--gold_preproc` do?
<br>
Regarding the hyperparameters for training how does the `batch size` work? Does it increase at each iteration? I don't understand how it fully works.
<br>
And, are the steps described [here](https://spacy.io/usage/training#tips) performed ("back-stage" ) when the `spacy train` command is run?

**Comment by adrianeboyd** [comment]
Some of the docs need to be updated here. (I don't know where `0, 1, -1` came from.)

`--use-gpu` refers to GPU ID (`-1` for CPU only). Iterations and epochs refer to the same thing here and the overall maximum is set by `--n-iter`.

`--n-examples` is the maximum number of documents for train or dev to use while training. (The console output will say "words", which is wrong.) It doesn't look like there's proper shuffling before the cutoff (the idea is save time by not reading in the full corpus), so which documents may depend on how the train and dev directories are traversed.

`--gold-preproc` is mainly for specific testing in scenarios where you want gold tokenization and gold sentence boundaries. If you use `--gold-preproc`, the parser won't learn how to split sentences. There really should be two separate options here. My recommendation: don't use this option. (You should try to have `"raw"` texts in your training data so you get an accurate picture of how well the models work in combination with spacy's tokenization, not just the provided tokenization.)

Most of those recommendations are used in `spacy train`. You can see all the settings in the script:

https://github.com/explosion/spaCy/blob/b9afcd56e3532125ec15f7d1f0825608c04835e3/spacy/cli/train.py
